packages/rum-core/src/domain/assembly.ts

```diff
@@ -97,6 +101,9 @@ export function startRumAssembly(
           ;(serverRumEvent.usr as RumEvent['usr']) = commonContext.user as User & Context
         }
         if (shouldSend(serverRumEvent, configuration.beforeSend, errorFilter)) {
+          if (isEmptyObject(serverRumEvent.context)) {
+            delete serverRumEvent.context
+          }
```

Comment:
removing the context when empty, is it an optimisation or is there more to it?

Reply:
It's to get the same behavior as before (which was an optimization, to avoid sending an unused property)